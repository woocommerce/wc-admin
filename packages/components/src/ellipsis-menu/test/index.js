/**
 * External dependencies
 */
import { render, fireEvent } from '@testing-library/react';

/**
 * Internal dependencies
 */
import EllipsisMenu from '../';

describe( 'EllipsisMenu', () => {
	it( 'renders correctly', () => {
		const { container } = render(
			<EllipsisMenu
				label={ 'foo' }
				renderContent={ () => <div>content</div> }
			/>
		);
		expect( container ).toMatchSnapshot();
	} );

	it( 'adds the passed in classname', () => {
		const { container } = render(
			<EllipsisMenu
				label={ 'foo' }
				className="custom-classname"
				renderContent={ () => <div>content</div> }
			/>
		);

		const menu = container.querySelector( '.custom-classname' );
		expect( menu ).toBeInTheDocument();
	} );

	it( 'should call onToggle when clicking on the ellipsis', () => {
		const onClickMock = jest.fn();
		const { getByTitle } = render(
			<EllipsisMenu
				label={ 'foo' }
				onToggle={ onClickMock }
				renderContent={ () => <div>content</div> }
			/>
		);

		fireEvent.click( getByTitle( 'foo' ) );
		expect( onClickMock ).toHaveBeenCalledTimes( 1 );
	} );

	it( 'should render content when clicking on the ellipsis', () => {
		const { getByTitle, getByText } = render(
			<EllipsisMenu
				label={ 'foo' }
				renderContent={ () => <div>content</div> }
			/>
		);

		fireEvent.click( getByTitle( 'foo' ) );
		expect( getByText( 'content' ) ).toBeInTheDocument();
	} );
} );
