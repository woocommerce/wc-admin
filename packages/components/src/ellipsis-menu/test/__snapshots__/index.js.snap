// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EllipsisMenu renders correctly 1`] = `
<div>
  <div
    class="woocommerce-ellipsis-menu"
  >
    <div
      class="components-dropdown"
    >
      <button
        aria-expanded="false"
        class="components-button woocommerce-ellipsis-menu__toggle has-text has-icon"
        title="foo"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="dashicon dashicons-ellipsis"
          focusable="false"
          height="20"
          role="img"
          viewBox="0 0 20 20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5 10c0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2 2 .9 2 2zm12-2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-7 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
        <svg
          aria-hidden="true"
          focusable="false"
          height="24"
          role="img"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 9v1.5h16V9H4zm12 5.5h4V13h-4v1.5zm-6 0h4V13h-4v1.5zm-6 0h4V13H4v1.5z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
